server/google_resources.py

```diff
@@ -0,0 +1,126 @@
+import googleapiclient.discovery
+import models.google
+import requests
+import flask
+import json
+import os
+
+# Configuration
+# GOOGLE_CLIENT_ID = os.environ.get("GOOGLE_CLIENT_ID", None)
+# GOOGLE_CLIENT_SECRET = os.environ.get("GOOGLE_CLIENT_SECRET", None)
+CLIENT_SECRETS_FILE = "client_secret.json"
+GOOGLE_DISCOVERY_URL = (
+    "https://accounts.google.com/.well-known/openid-configuration"
+)
+SCOPES = [
+  'https://www.googleapis.com/auth/gmail.send',
+  # 'https://www.googleapis.com/auth/userinfo.email',
+  # 'https://www.googleapis.com/auth/userinfo.profile',
+]
+API_SERVICE_NAME = 'gmail'
+API_VERSION = 'v1'
+
+def credentials_to_dict(credentials):
+  return {'token': credentials.token,
+          'refresh_token': credentials.refresh_token,
+          'token_uri': credentials.token_uri,
+          'client_id': credentials.client_id,
+          'client_secret': credentials.client_secret,
+          'scopes': credentials.scopes}
+
+class TestAPIRequest(Resource):
+  # def test_api_request(): # TODO: Change from google drive request.
+  def get(self): # TODO: Change from google drive request.
+    if 'credentials' not in flask.session:
+      return flask.redirect('authorize')
+
+    # Load credentials from the session.
+    credentials = google.oauth2.credentials.Credentials(
+        **flask.session['credentials'])
+
+    drive = googleapiclient.discovery.build(
+        API_SERVICE_NAME, API_VERSION, credentials=credentials)
+
+    files = drive.files().list().execute()
+
+    # Save credentials back to session in case access token was refreshed.
+    # ACTION ITEM: In a production app, you likely want to save these
+    #              credentials in a persistent database instead.
+    flask.session['credentials'] = credentials_to_dict(credentials)
+
+    return flask.jsonify(flask.session)
+
+class Authorize(Resource):
+  # def authorize():
+  def get(self):
+    # Create flow instance to manage the OAuth 2.0 Authorization Grant Flow steps.
+    flow = google_auth_oauthlib.flow.Flow.from_client_secrets_file(
+        CLIENT_SECRETS_FILE, scopes=SCOPES)
+
+    # The URI created here must exactly match one of the authorized redirect URIs
+    # for the OAuth 2.0 client, which you configured in the API Console. If this
+    # value doesn't match an authorized URI, you will get a 'redirect_uri_mismatch'
+    # error.
+    flow.redirect_uri = flask.url_for('oauth2callback', _external=True)
+
+    authorization_url, state = flow.authorization_url(
+        # Enable offline access so that you can refresh an access token without
+        # re-prompting the user for permission. Recommended for web server apps.
+        access_type='offline',
+        # Enable incremental authorization. Recommended as a best practice.
+        include_granted_scopes='true')
+
+    # Store the state so the callback can verify the auth server response.
+    flask.session['state'] = state
+
+    return flask.redirect(authorization_url)
```

Comment:
This is the reason you are getting a CORS error.

The front-end React app only is accepting JSON RESTful API responses, but this is a redirect. Your front-end app won't be able to accept redirects. 

Instead, you need to change the flow to something like this:
- this API returns a dictionary something like {url: authorization_url}
- the front-end would read that authorization_url and set it so when the Authorize Button is pressed, you redirect the application to the authorization_url

Now this also has implications on how we handle the redirection after the authorization URL is pressed. In this case again the result would go to the React app, so the React app must read the data from the parameters that it receives and send a JSON API request to the back-end manually with that information.